### ISSUE TITLE
[types] Model.create dosnt return array

### ISSUE THREAD

**Issue by hasezoey** [Automattic/mongoose#9817]
**Do you want to request a *feature* or report a *bug*?**
type-bug

**What is the current behavior?**
`await SomeModel.create([doc1, doc2])` dosnt return an array

**If the current behavior is a bug, please provide the steps to reproduce.**
code used:
```ts
// NodeJS: 14.15.1
// MongoDB: 4.2-bionic (Docker)
// import { getModelForClass, prop } from "@typegoose/typegoose"; // @typegoose/typegoose@7.4.5
import * as mongoose from "mongoose"; // mongoose@5.11.12

interface ISomeModel extends mongoose.Document {
  param1: string,
  param2: number,
}

const SomeSchema = new mongoose.Schema({ param1: String, param2: Number });
const SomeModel = mongoose.model<ISomeModel>("SomeModel", SomeSchema);

(async () => {
  await mongoose.connect(`mongodb://localhost:27017/`, { useNewUrlParser: true, dbName: "verifyMASTER", useCreateIndex: true, useUnifiedTopology: true });

  const [...docs] = await SomeModel.create([{ param1: "Hello", param2: 1 }]);
  const doc = await SomeModel.create({});


  await mongoose.disconnect();
})();
```

```json
{
    "compileOnSave": true,
    "typeAcquisition": {
        "enable": true
    },
    "compilerOptions": {
        "experimentalDecorators": true,
        "emitDecoratorMetadata": true,
        "declarationMap": true,
        "outDir": "lib",
        "moduleResolution": "node",
        "target": "es6",
        "module": "commonjs",
        "newLine": "LF",
        "sourceMap": true,
        "removeComments": true,
        "strict": true,
        "allowUnreachableCode": false,
        "pretty": true,
        "declaration": true,
        "lib": [
            "esnext"
        ]
    },
    "include": [
        "src/**/*"
    ]
}
```

**What is the expected behavior?**
that `await SomeModel.create([doc1,doc2])` returns an array

**What are the versions of Node.js, Mongoose and MongoDB you are using?**
mongoose 5.11.12
typescript 4.1.3

would have provided an pr, but didnt find an fix yet

**Comment by Duchynko** [comment]
The problem seems to be here

https://github.com/Automattic/mongoose/blob/c44d521e18d292817a37c9c1623721da52a249f6/index.d.ts#L634

```ts
/** Creates a new document or documents */
create<Z = T | DocumentDefinition<T>>(doc: Z): Promise<T>;
create<Z = T | DocumentDefinition<T>>(docs: Array<Z>, options?: SaveOptions): Promise<Array<T>>;
```

If you don't specify an options parameter (although it's optional...) it will assume that the return value is `T`.

Try to do this. It should help.
```ts
await SomeModel.create([{ param1: "Hello", param2: 1 }], {});
```

I'm not sure how to fix that though - my guess would be that it should work, but it doesn't. I'm not as experienced with type declarations.

**Comment by hasezoey** [comment]
thanks @Duchynko, do you know why this line is ignored then?

https://github.com/Automattic/mongoose/blob/c44d521e18d292817a37c9c1623721da52a249f6/index.d.ts#L635

Note: i already manually tried to set the return of this line to `Promise<Array<T>>` (which is i think the proper one for this line), but it didnt change what typescript will use
-> and i also tried switching line 634 and 635 - which didnt change the outcome (i also tried just commenting out the line 634, which also didnt change anything)

**Comment by Duchynko** [comment]
@hasezoey I'm not sure, hah. I also tried to play around with this and didn't figure it out.

https://github.com/Automattic/mongoose/blob/c44d521e18d292817a37c9c1623721da52a249f6/index.d.ts#L635

@vkarpov15 shouldn't the above overload return an array? I can create a PR if so...
```ts
create<Z = T | DocumentDefinition<T>>(...docs: Array<Z>): Promise<Array<T>>;
```

**Comment by vkarpov15** [comment]
@Duchynko you're right, it should. Passing a spread of docs means you get a spread in the callback, but you do get an array of docs if you use a promise. Feel free to put in a PR :+1:

**Comment by hasezoey** [comment]
actual issue should get fixed by #9890 (didnt test it yet)
